packages/marketplace/test/exchange/MatchOrders.behavior.ts

```diff
@@ -309,6 +312,88 @@ export function shouldMatchOrders() {
         expect(await ERC20Contract2.balanceOf(maker)).to.be.equal(20000000000);
         expect(await ERC20Contract2.balanceOf(taker)).to.be.equal(0);
       });
+      it('recipient test', async function () {
+        const recipientAddress = await recipient.getAddress();
+        const makerAssetForLeftOrder = await AssetERC20(
+          ERC20Contract,
+          10000000000
+        );
+        const takerAssetForLeftOrder = await AssetERC20(
+          ERC20Contract2,
+          20000000000
+        );
+        // partially filled Asset
+        const takerAssetForRightOrder = await AssetERC20(
+          ERC20Contract2,
+          10000000000
+        );
+        // partially filled Asset
+        const makerAssetForRightOrder = await AssetERC20(
+          ERC20Contract,
+          5000000000
+        );
+
+        // left order for partial fill
+        orderLeft = await OrderDifferentRecipient(
+          maker,
+          makerAssetForLeftOrder,
+          ZeroAddress,
+          takerAssetForLeftOrder,
+          recipientAddress,
+          1,
+          0,
+          0
+        );
+        // right order for partial fill
+        orderRight = await OrderDifferentRecipient(
+          taker,
+          takerAssetForRightOrder,
+          ZeroAddress,
+          makerAssetForRightOrder,
+          recipientAddress,
+          1,
+          0,
+          0
+        );
+
+        makerSig = await signOrder(orderLeft, maker, OrderValidatorAsAdmin);
+        takerSig = await signOrder(orderRight, taker, OrderValidatorAsAdmin);
+
+        expect(
+          await ExchangeContractAsUser.fills(hashKey(orderLeft))
+        ).to.be.equal(0);
+        expect(
+          await ExchangeContractAsUser.fills(hashKey(orderRight))
+        ).to.be.equal(0);
+        expect(await ERC20Contract.balanceOf(maker)).to.be.equal(10000000000);
+        expect(await ERC20Contract.balanceOf(taker)).to.be.equal(0);
+        expect(await ERC20Contract2.balanceOf(maker)).to.be.equal(0);
+        expect(await ERC20Contract2.balanceOf(taker)).to.be.equal(20000000000);
+
+        await ExchangeContractAsUser.matchOrders([
+          {
+            orderLeft,
+            signatureLeft: makerSig,
+            orderRight,
+            signatureRight: takerSig,
+          },
+        ]);
+
+        expect(
+          await ExchangeContractAsUser.fills(hashKey(orderLeft))
+        ).to.be.equal(10000000000);
+        expect(
+          await ExchangeContractAsUser.fills(hashKey(orderRight))
+        ).to.be.equal(5000000000);
+        expect(await ERC20Contract.balanceOf(maker)).to.be.equal(5000000000);
+        expect(await ERC20Contract.balanceOf(recipientAddress)).to.be.equal(
+          5000000000
+        );
+        expect(await ERC20Contract2.balanceOf(recipientAddress)).to.be.equal(
+          10000000000
+        );
+        expect(await ERC20Contract2.balanceOf(taker)).to.be.equal(10000000000);
+      });
```

Comment:
We should add a test that check if the field recipient is contained in the signature and so that the buyer cannot change the seller recipient

Reply:
and add tests to cover transferRoyalties & transferPercentage you missed in the code